packages/react-reconciler/src/ReactFiberScheduler.old.js

```diff
@@ -581,6 +584,14 @@ function commitPassiveEffects(root: FiberRoot, firstEffect: Fiber): void {
   if (!isBatchingUpdates && !isRendering) {
     performSyncWork();
   }
+
+  if (__DEV__) {
+    if (rootWithPendingPassiveEffects === root) {
```

Comment:
Only keep incrementing if flushing root's deferred effects schedules the same root again.